examples/stateful-workloads-valkey/main.tf

```diff
@@ -0,0 +1,251 @@
+
+  lifecycle {
+    replace_triggered_by = [azurerm_container_registry_task.this]
+  }
+}
+
+## Section to create the Azure Kubernetes Service
+######################################################################################################################
+module "default" {
+  source                    = "../.."
+  name                      = coalesce(var.cluster_name, module.naming.kubernetes_cluster.name_unique)
+  resource_group_name       = azurerm_resource_group.this.name
+  location                  = azurerm_resource_group.this.location
+  sku_tier                  = "Standard"
+  local_account_disabled    = false
+  node_os_channel_upgrade   = "NodeImage"
+  automatic_upgrade_channel = "stable"
+
+  default_node_pool = {
+    name                    = "systempool"
+    node_count              = 3
+    vm_size                 = "Standard_D2ds_v4"
+    os_type                 = "Linux"
+    auto_upgrade_channel    = "stable"
+    node_os_upgrade_channel = "NodeImage"
+    zones                   = [1, 2, 3]
+
+    addon_profile = {
+      azure_key_vault_secrets_provider = {
+        enabled = true
+      }
+    }
+    upgrade_settings = {
+      max_surge = "10%"
+    }
+  }
+
+  node_pools                = var.node_pools
+  oidc_issuer_enabled       = true
+  workload_identity_enabled = true
+  network_profile = {
+    network_plugin = "azure"
+  }
+  key_vault_secrets_provider = {
+    secret_rotation_enabled = true
+  }
+}
+
+## Section to assign the role to the kubelet identity
+######################################################################################################################
+resource "azurerm_role_assignment" "acr_role_assignment" {
+  principal_id         = module.default.kubelet_identity_id
+  scope                = module.avm_res_containerregistry_registry.resource_id
+  role_definition_name = "AcrPull"
+
+  depends_on = [module.avm_res_containerregistry_registry, module.default]
+}
+
+resource "random_password" "requirepass" {
+  length           = 16
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+  special          = true
+}
+
+resource "random_password" "primaryauth" {
+  length           = 16
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+  special          = true
+}
+
+resource "local_file" "valkey_password_file" {
+  filename = "/tmp/valkey-password-file.conf"
+  content  = <<EOF
+requirepass  ${coalesce(var.valkey_password, random_password.requirepass.result)}
+primaryauth  ${coalesce(var.valkey_password, random_password.primaryauth.result)}
+EOF
+}
+
+resource "azurerm_key_vault_secret" "valkey_password_file" {
+  key_vault_id = module.avm_res_keyvault_vault.resource_id
+  name         = "valkey-password-file"
+  value        = local_file.valkey_password_file.content
+}
+
+resource "null_resource" "cleanup" {
+  triggers = {
+    always_run = timestamp()
+  }
+
+  provisioner "local-exec" {
+    command = "rm /tmp/valkey-password-file.conf"
+  }
+}
```

Comment:
need suggestions on how we can do this otherwise this is the best way to manage this

Reply:
the resource `azurerm_key_vault_secret` in the `value` uses the content directly: `local_file.valkey_password_file.content`

why you need to create a file ?

maybe you could try to remove the `valkey_password_file` completely and use:

```
<<EOF
requirepass  ${coalesce(var.valkey_password, random_password.requirepass.result)}
primaryauth  ${coalesce(var.valkey_password, random_password.primaryauth.result)}
EOF
```

directly in the `value` argument of the `azurerm_key_vault_secret`.

please let me know if this works. thanks